src/Chainweb/Utils.hs

```diff
@@ -404,6 +407,17 @@ minusOrZero :: Ord a => Num a => a -> a -> a
 minusOrZero a b = a - min a b
 {-# INLINE minusOrZero #-}
 
+-- | Equivalent to V.thaw . V.fromList but by inspection probably faster.
+mutableVectorFromList
+    :: PrimMonad m
+    => [a]
+    -> m (MV.MVector (PrimState m) a)
+mutableVectorFromList as = do
+    vec <- MV.unsafeNew (length as)
+    forM_ (zip [0..] as) $ uncurry (MV.unsafeWrite vec)
```

Comment:
This traverses the list twice, which could be an issue for very large lists. For moderately sized lists it probably is fine.

Reply:
I'm fairly sure (though not familiar enough with how Vector does its fusion) that `V.fromList` does it as well, so at least it isn't a loss.